### PR TITLE
feat: Add chatGPT-RAG plugin

### PR DESCRIPTION
**Pull request by IkbelTalebHssan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,88 @@
-# hexabot-plugin-chatgpt
-The OpenAI ChatGPT Plugin for Hexabot Chatbot / Agent Builder to enable the LLM RAG Capability
+# ChatGPT RAG Plugin for Hexabot Chatbot Builder
+
+The ChatGPT RAG Plugin integrates OpenAI’s GPT-4 model with Hexabot chatbot workflows to enable Retrieval-Augmented Generation (RAG) capabilities. This plugin allows users to enrich their chatbot responses using relevant content retrieved from a content management system, making interactions more contextual and accurate.
```
Comment (marrouchi) [inline README.md:3]:
```suggestion
The ChatGPT Plugin integrates OpenAI’s ChatGPT model with Hexabot chatbot workflows to enable Retrieval-Augmented Generation (RAG) capabilities. This plugin allows users to enrich their chatbot responses using relevant content retrieved from a content management system, making interactions more contextual and accurate.
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,88 @@
-# hexabot-plugin-chatgpt
-The OpenAI ChatGPT Plugin for Hexabot Chatbot / Agent Builder to enable the LLM RAG Capability
+# ChatGPT RAG Plugin for Hexabot Chatbot Builder
```
Comment (marrouchi) [inline README.md:1]:
```suggestion
# ChatGPT Plugin for Hexabot Chatbot Builder
```

Can we also rename the file to chatgpt.plugin.ts ?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,88 @@
...
+
+The ChatGPT RAG Plugin integrates OpenAI’s GPT-4 model with Hexabot chatbot workflows to enable Retrieval-Augmented Generation (RAG) capabilities. This plugin allows users to enrich their chatbot responses using relevant content retrieved from a content management system, making interactions more contextual and accurate.
+
+[Hexabot](https://hexabot.ai/) is an open-source chatbot / agent solution that allows users to create and manage AI-powered, multi-channel, and multilingual chatbots with ease. If you would like to learn more, please visit the [official github repo](https://github.com/Hexastack/Hexabot/).
+
+## Prerequisites
+
+Before setting up the ChatGPT RAG Plugin, you will need to generate an API token from OpenAI.
```
Comment (marrouchi) [inline README.md:9]:
```suggestion
Before setting up the ChatGPT Plugin, you will need to generate an API token from OpenAI.
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,88 @@
...
+
+```sh
+cd ~/projects/Hexabot
+npm install
+```
+
+To install the ChatGPT RAG Plugin, run the following command:
+
+```sh
+npx hexabot install plugin Hexastack/hexabot-plugin-chatgpt-rag
```
Comment (marrouchi) [inline README.md:27]: This is the name of the repo : 
```suggestion
npx hexabot install plugin Hexastack/hexabot-plugin-chatgpt
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,88 @@
...
+## Installation
+
+First, navigate to your Hexabot project directory and make sure the dependencies are installed:
+
+```sh
+cd ~/projects/Hexabot
+npm install
+```
+
+To install the ChatGPT RAG Plugin, run the following command:
```
Comment (marrouchi) [inline README.md:24]:
```suggestion
To install the ChatGPT Plugin, run the following command:
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,88 @@
...
+
+```sh
+npx hexabot install plugin Hexastack/hexabot-plugin-chatgpt-rag
+```
+
+## Configuration
+
+The ChatGPT RAG Plugin provides several customizable settings that can be configured through the Hexabot admin interface:
```
Comment (marrouchi) [inline README.md:32]:
```suggestion
The ChatGPT Plugin provides several customizable settings that can be configured through the Hexabot admin interface:
```

File: README.md [hunk 7]
```diff
@@ -1,2 +1,88 @@
...
+CONTEXT: You are an AI Chatbot that works for Hexastack. Here is a description of the company: ...
+
+DOCUMENTS:
+DOCUMENT 0
+Title: Example Title
+Data: Example Data...
+RECENT MESSAGES:
+
+- Message 1: Hi, how can I assist you today?
+- Message 2: Can you explain more about Hexastack’s services?
```
Comment (marrouchi) [inline README.md:61]: Is this the way OpenAI ChatGPT recommends in their documentation ?

File: package.json [hunk 8]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "hexabot-chatgpt-rag",
```
Comment (marrouchi) [inline package.json:2]:
```suggestion
  "name": "hexabot-plugin-chatgpt",
```

File: package.json [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+  "version": "2.0.0",
+  "description": "The OpenAI ChatGPT Plugin for Hexabot Chatbot / Agent Builder to enable the LLM RAG Capability",
+  "main": "chatgpt-rag.plugin.ts",
```
Comment (marrouchi) [inline package.json:5]: We don't need this
```suggestion
```

**Review by marrouchi — CHANGES_REQUESTED** [review]
Looks neat 👏

File: README.md (hunk 2 above)
Comment (IkbelTalebHssan) [reply README.md]: Sure we can :heavy_check_mark:

File: README.md (hunk 4 above)
Comment (IkbelTalebHssan) [reply README.md]: Okay updated :heavy_check_mark: 
Thank you :pray:

File: README.md (hunk 7 above)
Comment (IkbelTalebHssan) [reply README.md]: This example inspired from the [documentation](https://platform.openai.com/docs/examples).

File: package.json (hunk 9 above)
Comment (IkbelTalebHssan) [reply package.json]: Okay removed :heavy_check_mark:

**Review by yassinedorbozgithub — APPROVED** [review]

File: README.md [hunk 10]
```diff
@@ -1,2 +1,96 @@
...
+DOCUMENTS:
+DOCUMENT 0
+Title: Example Title
+Data: Example Data...
+
+RECENT MESSAGES:
+
+1. User: Hi, how can I assist you today?
+2. User: Can you explain more about Hexastack’s services?
+
```
Comment (marrouchi) [inline README.md:66]:
```suggestion
```

**Review by marrouchi — CHANGES_REQUESTED** [review]

**Pull request merged**